src/components/app.tsx

```diff
@@ -99,34 +107,35 @@ export const App = () => {
     setPlaybackProgress(0);
   };
 
-  useEffect(() => {
+    useEffect(() => {
     // AudioContext is apparently unavailable in the node / jest environment.
     // So we bail out early, to prevent render test failure.
     if (!window.AudioContext) { return; }
-
     setupAudioContext(selectedSound);
-  }, [selectedSound]);
+  },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
It's very rare we really want to ignore this rule. So, I'd either put a comment that explains why it happened, or actually try to follow ESLint suggestions and update the array (which sometimes might lead to other changes, but in my personal experience - ESLint was always right :wink:). It seems to me that I'd rather do the latter in this case.

Reply:
>// Not including setupAudioContext() in dependency array; to avoid render loop.

If that's the reason, then the right solution is to wrap the `setupAudio()` function in `useCallback()` so that it's stable across renders.

Reply:
Right, I'd vote to remove `// eslint-disable-next-line react-hooks/exhaustive-deps` and use `useCallback`. It seems like nitpicking, but I just think it's a good practice and bugs related to `useEffect` can be awful to trace (not necessarily in this case, but generally). Personally, I've only seen one (half) valid reason to skip this ESLint check since I started using `useEffect`. But it was actually the opposite - I've added extra variables to the dependency array to trigger update of non-react code from it. I've never had to remove a suggested dependency from this array.